torch/nn/utils/spectral_norm.py

```diff
@@ -0,0 +1,117 @@
+"""
+Spectral Normalization from https://arxiv.org/abs/1802.05957
+"""
+import torch
+from torch.nn.parameter import Parameter
+
+
+def l2normalize(v, eps=1e-12):
```

Comment:
Can't you use instead [`torch.nn.functional.normalize`](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L2004)?

Reply:
I mimic the author's [implementation](https://github.com/pfnet-research/sngan_projection/blob/master/source/functions/max_sv.py#L5) where the denominator is `norm + eps`, not `max(norm, eps)`.

I'm not sure which is suitable so if you think `F.normalize` better, I'll replace.

Reply:
I think there is no real difference actually. This is there just to avoid division by zero.